commit f0e898398ce3f2342f9da61d18b00219a2874d15
Author: ners <ners@gmx.ch>
Date:   Mon Jun 10 11:53:28 2024 +0200

    display diff between old and new profiles

diff --git a/direnvrc b/direnvrc
index 68cef33..552b690 100644
--- a/direnvrc
+++ b/direnvrc
@@ -299,6 +299,8 @@ use_flake() {
       local tmp_profile_rc
       local tmp_profile="${layout_dir}/flake-tmp-profile.$$"
       if tmp_profile_rc=$(_nix print-dev-env --profile "$tmp_profile" "$@"); then
+        local old_store_path=$(find "${layout_dir}" -type l -name '*-profile-*' | head -n1 | xargs readlink)
+
         # If we've gotten here, the user's current devShell is valid and we should cache it
         _nix_clean_old_gcroots "$layout_dir"
 
@@ -320,6 +322,9 @@ use_flake() {
           flake_input_paths="${flake_input_paths/${store_path}/}"
         done
 
+        if [[ -n "${old_store_path}" ]]; then
+          nvd diff "${old_store_path}" "$(readlink $profile)"
+        fi
         _nix_direnv_info "Renewed cache"
       else
         # The user's current flake failed to evaluate,
@@ -478,11 +483,16 @@ use_nix() {
         --profile "$tmp_profile" \
         --impure \
         "${extra_args[@]}"); then
+        local old_store_path=$(find "${layout_dir}" -type l -name '*-profile-*' | head -n1 | xargs readlink)
         _nix_clean_old_gcroots "$layout_dir"
 
         echo "$tmp_profile_rc" >"$profile_rc"
         _nix_add_gcroot "$tmp_profile" "$profile"
         rm -f "$tmp_profile" "$tmp_profile"*
+
+        if [[ -n "${old_store_path}" ]]; then
+          nvd diff "${old_store_path}" "$(readlink $profile)"
+        fi
         _nix_direnv_info "Renewed cache"
       else
         _nix_direnv_warning "Evaluating current nix shell failed. Falling back to previous environment!"
