commit c5610414240b428e27ac3e56eb442b83115aa1b9
Author: ners <ners@gmx.ch>
Date:   Wed Feb 21 21:06:01 2024 +0100

    display diff between old and new profiles

diff --git a/direnvrc b/direnvrc
index 723f29c..7220b11 100644
--- a/direnvrc
+++ b/direnvrc
@@ -244,6 +244,7 @@ use_flake() {
       _nix_clean_old_gcroots "$layout_dir"
 
       # We need to update our cache
+      local old_store_path=$(find "${layout_dir}" -type l -name '*-profile-*' | head -n1 | xargs readlink)
       local tmp_profile="${layout_dir}/flake-profile.$$"
       local tmp_profile_rc
       tmp_profile_rc=$("${NIX_BIN_PREFIX}nix" print-dev-env \
@@ -269,6 +270,9 @@ use_flake() {
         _nix_add_gcroot "$path" "${flake_inputs}/${path##*/}"
       done
 
+      if [[ -n "${old_store_path}" ]]; then
+        nvd diff "${old_store_path}" "$(readlink $profile)"
+      fi
       log_status "nix-direnv: renewed cache"
     fi
   else
@@ -397,6 +401,7 @@ use_nix() {
      else
       _nix_clean_old_gcroots "$layout_dir"
 
+      local old_store_path=$(find "${layout_dir}" -type l -name '*-profile-*' | head -n1 | xargs readlink)
       local tmp_profile="${layout_dir}/flake-profile.$$"
       local tmp_profile_rc
 
@@ -423,6 +428,9 @@ use_nix() {
       echo "$tmp_profile_rc" > "$profile_rc"
       rm -f "$tmp_profile" "$tmp_profile"*
       _nix_add_gcroot "$drv" "$profile"
+      if [[ -n "${old_store_path}" ]]; then
+        nvd diff "${old_store_path}" "$(readlink $profile)"
+      fi
       log_status "nix-direnv: renewed cache"
 
     fi
